### PR TITLE
Bugfix: UI Pivot behaviour with new canvas refactor

### PR DESCRIPTION
**Pull request by AHinojal**
We have fixed the pivot! We can rotate a 2D object around a pivot despite the current position, scale, size and anchors ;)

I deleted the option in the editor panel to put pivot mode or center mode and if you want the center, you have to put (0.5,0.5) in the pivot (its by default in that position)
When scale and change the pivot, recompute the position to put the object in the center...because if you scale an object after changed the pivot, the object will have a little offset.
Also, you can't rotate, change the pivot and rotate again like in Unity...maybe we can put this option in the future if we have plenty of time, because honestly, I don't think we will use so much the pivot configuration.

If I see any bugs or something like that, put it Clickup as a hotfix! ;)

ezgif com-gif-maker (1)

### CODE REVIEW

File: Game/Assets/Shaders/uiShader.glsl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+uniform vec4 tintColor;
+
+out vec4 outColor;
+
+void main()
+{	
+	outColor = (hasDiffuse * texture2D(diffuse, uv0) + 1 - hasDiffuse) * inputColor * tintColor;
+}
+
+#endif
```
Comment (JaimeBea) [inline Game/Assets/Shaders/uiShader.glsl:37]: The .glsl shaders that are being added in this PR should be removed. They are deprecated.

**Review by JaimeBea — APPROVED** [review]

**Comment by JaimeBea** [comment]
Make sure to merge develop into the branch first.

**Review by JaimeBea — COMMENTED** [review]
You removed some shaders you didn't have to remove. Be careful.

**Comment by AHinojal** [comment]
Fix all problems with the shaders folder and glsl files in the last commit with the files that we have in the Shaders folder in the develop branch right now. ;)

**Review by FrancPS — APPROVED** [review]
Works as intended and no compilation warnings, good job.

As always, test a scene in GAME if you haven't.

I'll include the missing functionality in QA list.

**Comment by AHinojal** [comment]
> Works as intended and no compilation warnings, good job.
> 
> As always, test a scene in GAME if you haven't.
> 
> I'll include the missing functionality in QA list.

Luckily you have say that I should take a look testing in GAME and 2D option, because the fucking Bounding Box was wrong when rotate with a pivot different than (0.5,0.5) 😓🤦‍♂️ My fault!

When I fix this problem, I will request the review and put a comment with a proof

**Review by devildrake — APPROVED** [review]

**Review by FrancPS — CHANGES_REQUESTED** [review]
Fix errors appearing in GAME

File: Project/Source/Components/ComponentBoundingBox2D.cpp [hunk 2]
```diff
@@ -77,26 +66,29 @@ void ComponentBoundingBox2D::CalculateWorldBoundingBox(bool force) {
...
-		//worldAABB.maxPoint = position.xyz().Mul(float3(1.0f, -1.0f, 0.0f).Mul(screenFactor)) + float3(screenSize, 0.0f) / 2.0f
-							 //+ localAABB.maxPoint.Mul(float3(transform2d->GetSize(), 0.0f).Mul(transform2d->GetScale().xyz()).Mul(screenFactor));
-
 #if GAME
 		float2 windowPos = float2(App->window->GetPositionX(), App->window->GetPositionY());
-		worldAABB.minPoint += windowPos;
+		//worldAABB.minPoint += windowPos;
 		//worldOBB.minPoint += windowPos;
-		worldAABB.maxPoint += windowPos;
+		//worldAABB.maxPoint += windowPos;
```
Comment (JaimeBea) [inline Project/Source/Components/ComponentBoundingBox2D.cpp:91]: If this is not necessary anymore it could be removed. Check if it works in Game.

**Review by JaimeBea — APPROVED** [review]

**Review by devildrake — CHANGES_REQUESTED** [review]
There appears to be something wrong with the scaling when the pivot X is not 0.5, from gameplay it appears we only need the pivot to work properly on Y axis, which it does, but this is still a problem

https://user-images.githubusercontent.com/23354524/120064624-79101f00-c06d-11eb-9709-bcb4e64e0cb8.mp4

**Comment by AHinojal** [comment]
Well, I reopen the PR fixing the pivot for this last sprint! I tested in Debug, Release and Game the scene that I put here in this message.
You can scale the object from a pivot and the bounding box works correctly (to @devildrake)
Also I put an checkbox in the inspector in Canvas & BoundingBox2D components to draw the object's gizmo in 3D.

Bud luck for us, fix the bounding box (rotate behaviour specially) its more difficult that I thought...We will have to transform the boundingbox from AABB2D to a OBB (OBB2D's implemetation is empty xD) and like its not our priority and right now this feature we dont need it, we can try to fix it later.

Download TestScene: 
[TEST_PivotingScene.zip](https://github.com/PenteractStudios/Tesseract/files/6583697/TEST_PivotingScene.zip)

ezgif com-gif-maker (4)
ezgif com-gif-maker (2)

File: Project/Source/Components/UI/ComponentTransform2D.cpp [hunk 3]
```diff
@@ -16,6 +16,7 @@
 #include "Math/TransformOps.h"
 
 #include "Utils/Leaks.h"
+#include <Utils/Logging.h>
```
Comment (JaimeBea) [inline Project/Source/Components/UI/ComponentTransform2D.cpp:19]: Leaks should be the last include

**Review by JaimeBea — DISMISSED** [review]

File: Project/Source/Components/UI/ComponentTransform2D.cpp (hunk 3 above)
Comment (AHinojal) [reply Project/Source/Components/UI/ComponentTransform2D.cpp]: Perfect! Fix it in the last commit ;)

**Review by JaimeBea — APPROVED** [review]

**Review by devildrake — APPROVED** [review]
Wonderful! Good job guys!

**Review by FrancPS — APPROVED** [review]
There is a minor issue about scaling UI elements and translating the scale to its children BB.

Added as another task is clickup since its a minor issue, and a feature that will probably not be used.

**Pull request merged**